### DISCUSSION TITLE
zustand and local storage

### DISCUSSION THREAD

**Discussion by yassine-ramla** [pmndrs/zustand#2522] [Q&A]
i have the following store:
 ```ts
export const useGameStore = create<Game & GameActions>()(
  persist(
    (set) => ({
      theme: "icons",
      updateTheme: (theme) => set(() => ({ theme: theme })),
    }),
    {
      name: "game",
    }
  )
);
```
and the following component:
```ts
import RadioButton from "@/components/radio-button/radio-button";
import { useGameStore } from "@/store/game";

export default function Home() {
  const {
    theme,
    updateTheme,
  } = useGameStore();

  return (
    <main>
            <RadioButton
              id="numbers"
              label="Numbers"
              name="theme"
              value="numbers"
              updateStore={updateTheme}
              checked={theme === "numbers"}
            />
            <RadioButton
              id="icons"
              label="Icons"
              name="theme"
              value="icons"
              updateStore={updateTheme}
              checked={theme === "icons"}
            />
    </main>
  );
}
```
the problem is even when i change the value of the "theme" to "numbers" by checking radio button and it's value changes in the local storage, when i refresh the page, it takes it default value and the condition ''theme === "numbers"'' is false, i think it should look in the local storage first, how can i fix that?

**Comment by dbritto-dev** [comment]
@yassine-ramla would you mind sharing us a minimal repro on stackblitz?

**Reply by yassine-ramla** [reply]
[here](https://stackblitz.com/edit/stackblitz-starters-lp1tln?description=The%20React%20framework%20for%20production&file=app%2Fpage.tsx&title=Next.js%20Starter) it is
i'm using firefox, the default value of "theme" is "icons", when i check the "numbers" radio button, the value of "theme" is updated to "numbers" and stored in the local storage, when i refresh, it takes its default value and "theme === 'numbers'" is false even that the value of "theme" is "numbers" in the local storage
